src/test/java/org/graylog/integrations/notifications/types/SlackEventNotificationTest.java

```diff
@@ -161,26 +152,83 @@ public void getCustomMessageModel() {
     }
 
 
-    @Test (expected = PermanentEventNotificationException.class)
+    @Test
+    @Ignore("Disabled, unable to test this method, since it return void")
```

Comment:
This does not make sense.  There are plenty of ways to meaningfully test a method that returns void.  Unit testing is not just about ensuring a method returns the expected value.  For methods like `execute`, whose job is to coordinate the action, you can validate that the right dependencies are being invoked with the right parameters.  You can validate that the error handling goes according to plan ("when dependency X throws a NPE, my method should throw a PermanentEventNotificationException", things like that).

If you need examples, feel free to look at [PagerDutyNotifiationTest](https://github.com/Graylog2/graylog-plugin-integrations/blob/master/src/test/java/org/graylog/integrations/pagerduty/PagerDutyNotificationTest.java)

The only time we should be removing unit tests is when we're removing the functionality they tested.

Reply:
Thanks for the example.